Implemented Delete Task Feature

faysal-alam-sazim opened the pull request:
### Description of Change
- Created delete task function in localstorage.ts
- Handled delete task button click
- Added alert before and after deletion
### Associated Ticket Link: 
- https://trello.com/c/JysOOXd1
### Related Pull Request: N/A
### Screenshots / Screen Recordings: 
<img width="1440" alt="Screen Shot 2024-07-11 at 3 28 22 PM" src="https://github.com/faysal-alam-sazim/fancy-todo-app/assets/174481120/86a2fa06-e865-4a79-ad11-a03e808c24cb">
<img width="1440" alt="Screen Shot 2024-07-11 at 3 28 30 PM" src="https://github.com/faysal-alam-sazim/fancy-todo-app/assets/174481120/b402c1fe-b204-4e0f-9ebb-475374a722d4">

faysal-alam-sazim committed:
Project setup

- Created React+Vite with TypeScrip project
- Configured Mantine UI library

faysal-alam-sazim committed:
Project setup

- Created React+Vite with TypeScrip project
- Configured Mantine UI library

faysal-alam-sazim committed:
Project setup

- Created React+Vite with TypeScrip project
- Configured Mantine UI library

faysal-alam-sazim committed:
Project setup

- Created React+Vite with TypeScrip project
- Configured Mantine UI library

faysal-alam-sazim committed:
Implemented MenuBar

- Added Title
- Added Create Task button
- Added Filter option
- Added Clear tasks button

faysal-alam-sazim committed:
Implemented MenuBar

- Added Title
- Added Create Task button
- Added Filter option
- Added Clear tasks button

faysal-alam-sazim committed:
Deleted Unnecessary Files

- Deleted assets folder as we are not using any

faysal-alam-sazim committed:
Deleted Unnecessary Files

- Deleted assets folder as we are not using any

faysal-alam-sazim committed:
Deleted Unnecessary Files

- Deleted assets folder as we are not using any

faysal-alam-sazim committed:
Deleted Unnecessary Files

- Deleted assets folder as we are not using any

faysal-alam-sazim committed:
chore(FTDA): Rebased with main and deleted commented code

faysal-alam-sazim committed:
feat(FTDA): Added Create Task Form

Implemented create task and saved task to localstorage. Created a modal consisting the form inputs, and written four functions in localstorage.ts

faysal-alam-sazim committed:
feat(FTDA): Added id & priority to each task

faysal-alam-sazim committed:
feat(FTDA): Added id & priority to each task

faysal-alam-sazim committed:
chore(FTDA): Fix DateInput css issue

faysal-alam-sazim committed:
chore(FTDA): Fix DateInput css issue

faysal-alam-sazim committed:
chore(FTDA): Add self closingtags for MenuBar

faysal-alam-sazim committed:
feat(FTDA): Display task

faysal-alam-sazim committed:
chore(FTDA): Add tailwindcss

faysal-alam-sazim committed:
feat(FTDA): add display task feature

faysal-alam-sazim committed:
chore(FTDA): Update folder structure & repositioned useEffect hook

faysal-alam-sazim committed:
feat(FTDA): Added Edit Task

faysal-alam-sazim committed:
feat(FTDA): Delete Task Feature

faysal-alam-sazim committed:
feat(FTDA): Delete Task Feature

ishmam-iqbal-sazim commented on src/localstorage/localstorage.ts:
```diff
@@ -33,10 +33,22 @@ function updateTaskInLocalStorage(task: Task) {
+    (item: Task) => Number(item.id) !== Number(task.id)
+  );
+  const tasksAfterDelete = [...filteredPrevTasks];
+  localStorage.removeItem("tasks");
+  localStorage.setItem("tasks", JSON.stringify(tasksAfterDelete));
```
Left a comment in another PR as well. No need to remove prior to setting tasks.

ishmam-iqbal-sazim reviewed (approved):
Approved pending comments.

faysal-alam-sazim committed:
chore(FTDA): commit local changes

faysal-alam-sazim committed:
Added Edit Task Feature (#6)

* Project setup

- Created React+Vite with TypeScrip project
- Configured Mantine UI library

* Implemented MenuBar

- Added Title
- Added Create Task button
- Added Filter option
- Added Clear tasks button

* Deleted Unnecessary Files

- Deleted assets folder as we are not using any

* Project setup

- Created React+Vite with TypeScrip project
- Configured Mantine UI library

* chore(FTDA): Removed unnecessary files

* Implemented MenuBar

- Added Title
- Added Create Task button
- Added Filter option
- Added Clear tasks button

* Deleted Unnecessary Files

- Deleted assets folder as we are not using any

* chore(FTDA): Deleted src/index.css

* chore(FTDA): Rebased with main and deleted commented code

* chore(FTDA): added .vite in .gitignore

* feat(FTDA): Added Create Task Form

Implemented create task and saved task to localstorage. Created a modal consisting the form inputs, and written four functions in localstorage.ts

* feat(FTDA): Added id & priority to each task

* chore(FTDA): Fix DateInput css issue

* Project setup

- Created React+Vite with TypeScrip project
- Configured Mantine UI library

* Deleted Unnecessary Files

- Deleted assets folder as we are not using any

* Project setup

- Created React+Vite with TypeScrip project
- Configured Mantine UI library

* Deleted Unnecessary Files

- Deleted assets folder as we are not using any

* chore(FTDA): Removed unnecessary files

* chore(FTDA): Add self closingtags for MenuBar

* feat(FTDA): Added id & priority to each task

* chore(FTDA): Fix DateInput css issue

* feat(FTDA): Display task

* chore(FTDA): Add tailwindcss

* feat(FTDA): add display task feature

* chore(FTDA): Update folder structure & repositioned useEffect hook

* feat(FTDA): Added Edit Task

* chore(FTDA): Update necessary changes from review

* chore(FTDA): Update necessary changes from review

* feat(FTDA): Add edit task feature

* chore(FTDA): Update modal title

* chore(FTDA): Update changes from review

faysal-alam-sazim committed:
Merge branch 'main' of https://github.com/faysal-alam-sazim/fancy-todo-app

faysal-alam-sazim committed:
chore(FTDA): Apply changes from review

faysal-alam-sazim committed:
chore(FTDA):resolve merge conflicts

faysal-alam-sazim committed:
Merge branch 'edit-task' into delete-task
